packages/grid/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -498,6 +504,15 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
     return -1;
   }, [cellFocus, currentPage.rows]);
 
+  const handleRowMouseEnter = useEventCallback<GridEventListener<'rowMouseEnter'>>((params) => {
+    setHoveredRowId(params.id ?? null);
+  });
+  const handleRowMouseLeave = useEventCallback<GridEventListener<'rowMouseLeave'>>(() => {
+    setHoveredRowId(null);
+  });
+  useGridApiEventHandler(apiRef, 'rowMouseEnter', handleRowMouseEnter);
+  useGridApiEventHandler(apiRef, 'rowMouseLeave', handleRowMouseLeave);
```

Comment:
If we're using `useGridApiEventHandler`, we don't need `useEventCallback` because it already does the same thing as `useEventCallback`.